src/cmd/flux-start.c

```diff
@@ -336,7 +336,7 @@ static void state_cb (flux_subprocess_t *p, flux_subprocess_state_t state)
                 log_msg ("%d (pid %d) %s", cli->rank, pid, strsignal (SIGCONT));
             else if (WIFSIGNALED (status))
                 log_msg ("%d (pid %d) %s", cli->rank, pid, strsignal (WTERMSIG (status)));
-            else if (WIFEXITED (status))
+            else if (WIFEXITED (status) && WEXITSTATUS (status) != 0)
                 log_msg ("%d (pid %d) exited with rc=%d", cli->rank, pid, WEXITSTATUS (status));
```

Comment:
Hm, I had thought you'd already fixed this. Maybe in an existing PR that hasn't been merged?

While you are in here, the tests for `WIFSTOPPED()` (actually erroneously written here as `WIFSIGNALED()`) and `WIFCONTINUED()` should be removed. This code is only executed under `FLUX_SUBPROCESS_EXITED` so those states will never be present.

Reply:
Oh, hmm, I thought I had fixed it the way you asked before too....  I cherry picked that from another PR thinking it would get in quicker if I tacked it on here.  I'll revist, thanks for catching.